optimum/habana/accelerate/data_loader.py

```diff
@@ -1,18 +1,269 @@
+import math
 from typing import Callable, List, Optional, Union
 
 import torch
 from accelerate.data_loader import (
     _PYTORCH_DATALOADER_KWARGS,
     BatchSamplerShard,
-    DataLoaderDispatcher,
     DataLoaderShard,
+    DataLoaderStateMixin,
     IterableDatasetShard,
     SeedableRandomSampler,
 )
-from accelerate.utils import RNGType
+from accelerate.state import GradientState
+from accelerate.utils import (
+    RNGType,
+    concatenate,
+    find_batch_size,
+    get_data_structure,
+    is_torch_version,
+    send_to_device,
+    slice_tensors,
+)
+from accelerate.utils.operations import _gpu_broadcast, is_tensor_information, recursively_apply
 from torch.utils.data import BatchSampler, DataLoader, IterableDataset
 
-from .state import GaudiAcceleratorState
+from .state import GaudiAcceleratorState, GaudiPartialState
+from .utils import GaudiDistributedType
+
+
+def initialize_tensors(data_structure):
+    """
+    Recursively initializes tensors from a nested list/tuple/dictionary of [`~utils.TensorInformation`].
+
+    Returns:
+        The same data structure as `data` with tensors instead of [`~utils.TensorInformation`].
+    """
+
+    def _initialize_tensor(tensor_info):
+        return torch.zeros(*tensor_info.shape, dtype=tensor_info.dtype)
+
+    return recursively_apply(_initialize_tensor, data_structure, test_type=is_tensor_information)
+
+
+def broadcast(tensor, from_process: int = 0):
+    """
+    Recursively broadcast tensor in a nested list/tuple/dictionary of tensors to all devices.
+
+    Args:
+        tensor (nested list/tuple/dictionary of `torch.Tensor`):
+            The data to gather.
+        from_process (`int`, *optional*, defaults to 0):
+            The process from which to send the data
+
+    Returns:
+        The same data structure as `tensor` with all tensors broadcasted to the proper device.
+    """
+    if GaudiPartialState().distributed_type in [GaudiDistributedType.MULTI_HPU, GaudiDistributedType.DEEPSPEED]:
+        return _gpu_broadcast(tensor, src=from_process)
+    return tensor
+
+
+def broadcast_object_list(object_list, from_process: int = 0):
+    """
+    Broadcast a list of picklable objects form one process to the others.
+
+    Args:
+        object_list (list of picklable objects):
+            The list of objects to broadcast. This list will be modified inplace.
+        from_process (`int`, *optional*, defaults to 0):
+            The process from which to send the data.
+
+    Returns:
+        The same list containing the objects from process 0.
+    """
+    if GaudiPartialState().distributed_type in [GaudiDistributedType.MULTI_HPU, GaudiDistributedType.DEEPSPEED]:
+        torch.distributed.broadcast_object_list(object_list, src=from_process, device="hpu")
+    return object_list
+
+
+class GaudiDataLoaderDispatcher(DataLoader, DataLoaderStateMixin):
```

Comment:
The only changes are the used of the methods above and not the ones of Accelerate right?

Reply:
yes